bip-0054/test_vectors/txsize.json

```diff
@@ -0,0 +1,62 @@
+[
+    {
+        "tx": "0200000001827da3d85a6547d6b03662d2cb86982d655a6f390547285a3bf9ec9f28e0c8831500000000ffffffff0100000000000000000300515200000000",
+        "valid": true,
+        "comment": "A 63-byte legacy transaction."
+    },
+    {
+        "tx": "02000000000101827da3d85a6547d6b03662d2cb86982d655a6f390547285a3bf9ec9f28e0c8831500000000ffffffff01000000000000000000010000000000",
+        "valid": true,
+        "comment": "A 61-byte legacy transaction with a witness."
+    },
+    {
+        "tx": "0200000001827da3d85a6547d6b03662d2cb86982d655a6f390547285a3bf9ec9f28e0c8831500000000ffffffff010000000000000000040051525400000000",
+        "valid": false,
+        "comment": "A 64-byte legacy transaction (4 bytes in spk)."
+    },
+    {
+        "tx": "0200000001827da3d85a6547d6b03662d2cb86982d655a6f390547285a3bf9ec9f28e0c883150000000403424242ffffffff010040075af07507000000000000",
+        "valid": false,
+        "comment": "A 64-byte legacy transaction (4 bytes in scriptsig)."
+    },
+    {
+        "tx": "0200000001827da3d85a6547d6b03662d2cb86982d655a6f390547285a3bf9ec9f28e0c8831500000000ffffffff01000000000000000005005152545800000000",
+        "valid": true,
+        "comment": "A 65-byte legacy transaction."
+    },
+    {
+        "tx": "02000000000101827da3d85a6547d6b03662d2cb86982d655a6f390547285a3bf9ec9f28e0c8831500000000ffffffff01000000000000000004005152540108213245576281941200000000",
+        "valid": false,
+        "comment": "A 64-byte Segwit transaction."
```

Comment:
I’m not sure if there is a coverage for a 64-byte Taproot transaction. Not sure if it’s either strictly necessary as Taproot transactions are Segwit transactions, even if the other way isn’t necessarily true.

There is no change with Segwit transactions, but having cases for a Taproot + annex (empty) and a Taproot + annex (full) to be sure that the 64-byte verification logic is checking the code correctly, can be valuable in my opinion.